_posts/2024-02-01-testing-with-wildfly-glow.adoc

```diff
@@ -0,0 +1,180 @@
+WildFly server in order to test the applications you are developing for WildFly.
+
+Note: reading this http://XXXX[blog post] that provides detailed information on WildFly provisioning in general and WildFly Glow in particular would be 
+a good pre-requisite prior to this blog post.
+
+The main goal of the WildFly Glow project is to help you produce a trimmed server that will properly execute your deployments.
+
+The trimmed WildFly server can be produced to run your tests or execute your final deployments. This blog post focus on the testing phase.
+
+== Provisioning approaches for your tests
+
+To install a server to execute your tests, you can:
+
+* https://www.wildfly.org/downloads/[Download] and install a WildFly distribution. 
+This approach is outside the scope of executing tests with a trimmed server. This would apply when execution your applications inside a full 
+WildFly server.
+
+* Provisioning a WildFly server using the WildFly Maven Plugin (with WildFly Glow support enabled) during test execution. 
+This approaches addresses simple applications composed of a single deployment. 
+
+* Use the https://github.com/wildfly/wildfly-arquillian[WildFly Arquillian Adapter] and WildFly Glow to provision a server to 
+test more complex use-cases.
+
+Note: The server used by the tests should contain the same content than the one used in production (with some configuration adjustments 
+to cope with each context). So testing with a full WildFly distribution when using a trimmed server in production will hide issues. 
+We strongly advise that you use provisioning in the first place and execute tests with a trimmed server.
+
+Ok, so you are ready to provision (or you already do) a trimmed server, but what is the provisioning configuration you need 
+to provide to the provisioning tooling to produce a WildFly server that will properly execute your tests?
+
+== Testing using the WildFly Maven Plugin to provision a server
+
+That is the simplest approach that applies to a single application deployed into WildFly. The WildFly quickstarts 
+cover this approach with the `provisioned-server` profile. Each quickstart defines this profile as you can find in the quickstarts https://github.com/wildfly/quickstart[source code].
+
+The `package` goal of the https://github.com/wildfly/wildfly-maven-plugin[WildFly Maven Plugin] scans the application, 
+discovers the provisioning configuration then provisions the server and deploy the application. 
+The server is started (`wildfly:start` goal) then the tests are executed to interact with the application deployed in the server.
+
+In some cases you need to run tests from inside the server or you want to tests multiple deployments, 
+that is where the WildFly Arquillian Adapter is needed.
+
+== Testing using the WildFly Arquillian Adapter
+
+=== Arquillian deployments
+
+When using Arquillian, the deployments to be deployed to the WildFly server are created by static Java methods 
```

Comment:
Think point out that these methods are in the test class, so

...methods in the test classes annotated with....